File: PreUtil/src/output/python/easy/NAryTreePreorderTraversal.py
#https://leetcode.com/problems/n-ary-tree-preorder-traversal/
#Given the root of an n-ary tree, return the preorder traversal of its nodes' values.
#Nary-Tree input serialization is represented in their level order traversal. Each group of children is separated by the null value (See examples)


"""
# Definition for a Node.
class Node(object):
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution(object):
    def preorder(self, root):
        """
        :type root: Node
        :rtype: List[int]
        """ 
        if root==None:
            return []
        if root.children==None:
            return [root.val]
        res=[root.val]
        for child in root.children:
            res.extend(self.preorder(child))
        return res
        

    

File: Tool/PreUtil/src/output/javascript/easy/NAryTreePreorderTraversal.js
/**https://leetcode.com/problems/n-ary-tree-preorder-traversal/ */
//Given the root of an n-ary tree, return the preorder traversal of its nodes' values.
//Nary-Tree input serialization is represented in their level order traversal. Each group of children is separated by the null value (See examples)


/**
 * // Definition for a Node.
 * function Node(val, children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */

/**
 * @param {Node|null} root
 * @return {number[]}
 */
var preorder = function(root) {
    let result = [];
    let stack = [];
    if(root != null){
        stack.push(root);
    }
    while(stack.length > 0){
        let current = stack.pop();
        result.push(current.val);
        for(let i = current.children.length - 1; i >= 0; i--){
            stack.push(current.children[i]);
        }
    }
    return result;
    
};

    